### PR TITLE
BO - Orders - Credit Slips - Sort & Pagination Credit Slips

### PR DESCRIPTION
**Pull request by cfarhani06**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | BO - Orders - Credit Slips - Sort & Pagination Credit Slips
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| Related PRs       | N/A
| How to test?      | TEST_PATH='functional/BO/02*/03*/02*' HEADLESS='false' npm run test:specific
| Possible impacts? | N/A

### CODE REVIEW

**Comment by Progi1984** [comment]
@cfarhani06 Thank you for your contribution. Could you fix lint errors ?

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+const loginCommon = require('@commonTests/BO/loginBO');
+const {createOrderByCustomerTest} = require('@commonTests/FO/createOrder');
+
+// Import BO pages
+const dashboardPage = require('@pages/BO/dashboard');
+const ordersPage = require('@pages/BO/orders/index');
+const orderPageTabListBlock = require('@pages/BO/orders/view/tabListBlock');
+const orderPageProductsBlock = require('@pages/BO/orders/view/productsBlock');
+const creditSlipsPage = require('@pages/BO/orders/creditSlips/index');
+// const creditSlipsListingPage = require('@pages/BO/catalog/products/index');
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]: Please delete the commented line

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+const numberOfOrderToCreate = 11;
+
+/*
+Pre-condition:
+- Create order in FO
+Scenario:
+- Create credit slip on the created order
+- Sort (by ID, Date and OrderID) and Pagination of Credit Slips
+ */
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]:
```suggestion
/*
Pre-condition:
- Create 11 orders in FO
- Create credit slip for each order
Scenario:
- Go to Orders > credit slips page
- Sort credit slips table by (ID, order ID, date issued)
- Pagination next and previous
 */
```

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+  it('should login in BO', async function () {
+    await loginCommon.loginBO(this, page);
+  });
+
+  for (let i = 0; i < numberOfOrderToCreate; i++) {
+    // Pre-condition: Create order in FO
+    createOrderByCustomerTest(orderByCustomerData, `baseContext_${i}`);
+
+    // eslint-disable-next-line no-loop-func
+    describe('Create Credit slip', async () => {
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]:
```suggestion
    describe(`Create Credit slip n°${i + 1}`, async () => {
```

File: tests/UI/pages/BO/orders/creditSlips/index.js [hunk 4]
```diff
@@ -26,6 +26,20 @@ class CreditSlips extends BOBasePage {
...
+    this.sortColumnSpanButton = column => `${this.sortColumnDiv(column)} span.ps-sort`;
+
+    // Pagination selectors
+    this.paginationBlock = '.pagination-block';
+    this.paginationLimitSelect = '#paginator_select_page_limit';
+    this.paginationLabel = `${this.creditSlipGridPanel} .col-form-label`;
+    this.paginationNextLink = `${this.creditSlipGridPanel} #pagination_next_url`;
+    this.paginationPreviousLink = `${this.creditSlipGridPanel} .pagination .previous a.page-link`;
+
+    // ///
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/orders/creditSlips/index.js]:
```suggestion
```

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js (hunk 1 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]: Done

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js (hunk 2 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]: Done

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js (hunk 3 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]: Done

File: tests/UI/pages/BO/orders/creditSlips/index.js (hunk 4 above)
Comment (cfarhani06) [reply tests/UI/pages/BO/orders/creditSlips/index.js]: Done

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+
+// Import data
+const {PaymentMethods} = require('@data/demo/paymentMethods');
+const {DefaultCustomer} = require('@data/demo/customer');
+const {Statuses} = require('@data/demo/orderStatuses');
+
+// Import test context
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_orders_creditSlips_sortAndPaginationCreditSlip';
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]:
```suggestion
const baseContext = 'functional_BO_orders_creditSlips_sortAndPaginationCreditSlips';
```

**Review by Progi1984 — COMMENTED** [review]
A small comment and it's an approval :tada:

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js (hunk 5 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]: Done

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @cfarhani06

**Pull request merged**